searx/plugins/limiter.py

```diff
@@ -0,0 +1,139 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Some bot protection / rate limitation
+
+pip install redis
+
+enable the plugin in settings.yml:
+* server.limiter: true
+* set redis.host, redis.port & redis.password
+"""
+
+import hmac
+import re
+from flask import request
+
+
+name = "Request limiter"
+description = "Limit the number of request"
+default_on = False
+preference_section = 'service'
+
+
+re_bot = re.compile(
+    r'([Cc][Uu][Rr][Ll]|[wW]get|Scrapy|splash|JavaFX|FeedFetcher|python-requests|Go-http-client|Java|Jakarta|okhttp|'
+    + r'HttpClient|Jersey|Python|libwww-perl|Ruby|SynHttpClient|UniversalFeedParser|Googlebot|GoogleImageProxy|'
+    + r'bingbot|Baiduspider|yacybot|YandexMobileBot|YandexBot|Yahoo! Slurp|MJ12bot|AhrefsBot|archive.org_bot|msnbot|'
+    + r'MJ12bot|SeznamBot|linkdexbot|Netvibes|SMTBot|zgrab|James BOT|Sogou|Abonti|Pixray|Spinn3r|SemrushBot|Exabot|'
+    + r'ZmEu|BLEXBot|bitlybot)')
+
+
+def is_accepted_request(get_aggregation_count):
+    # pylint: disable=too-many-return-statements
+    user_agent = request.headers.get('User-Agent')
+    x_forwarded_for = request.headers.get('X-Forwarded-For')
+
+    if request.path == '/image_proxy':
+        if re_bot.match(user_agent):
+            return False
+        return True
```

Comment:
> It is based on the same rules than https://github.com/searxng/searxng-docker

IMO we need a solution that is configurable, e.g. like filtron.

Reply:
> IMO we need a solution that is configurable

I agree, that's the issue of this PR
But what is the configuration is a python script?
I mean allow to replace
`limiter: true` the settings.yml, 
by `limiter: the_path_of_a_python_script.py`

This Python script has to define one function `is_accepted_request` and receive a parameter the `get_aggregation_count` function.

We can define a default function:
* `limiter: false`: the limiter is disabled.
* `limiter: true`: the default configuration (the one recommended for public instances. New compare to filtron: we can upgrade the configuration).
* `limiter: /opt/searxng/my_limit_conf.py`: a custom configuration that replace the existing one. The default configuration can be used a template.